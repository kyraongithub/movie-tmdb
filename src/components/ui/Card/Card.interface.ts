import type { MovieInterface } from '../../../interfaces/Movie.interface';

export interface CardPropsInterface {
  movie: MovieInterface;
  id: number;
  className?: string;
}
