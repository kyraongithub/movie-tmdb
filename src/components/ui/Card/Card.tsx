import type { CardPropsInterface } from './Card.interface';

const Card = (props: CardPropsInterface) => {
  const { title } = props;
  return <div>{title}</div>;
};

export default Card;
