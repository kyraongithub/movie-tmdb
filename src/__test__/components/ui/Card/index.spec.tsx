import { render } from '@testing-library/react';
import Card from '../../../../components/ui/Card';

describe('Card', () => {
  it('render Card component', () => {
    const movie = {
      adult: false,
      backdrop_path: '/eDBZN0TbWkxoAB0qIDFagVcPPTN.jpg',
      genre_ids: [27, 14, 35, 12],
      id: 1153714,
      original_language: 'en',
      original_title: 'Death of a Unicorn',
      overview:
        'A father and daughter accidentally hit and kill a unicorn while en route to a weekend retreat, where his billionaire boss seeks to exploit the creatureâ€™s miraculous curative properties.',
      popularity: 186.0114,
      poster_path: '/lXR32JepFwD1UHkplWqtBP1K79z.jpg',
      release_date: '2025-03-27',
      title: 'Death of a Unicorn',
      video: false,
      vote_average: 6.4,
      vote_count: 173,
    };

    const { container } = render(<Card movie={movie} id={0} />);
    expect(container).toMatchSnapshot();
  });
});
