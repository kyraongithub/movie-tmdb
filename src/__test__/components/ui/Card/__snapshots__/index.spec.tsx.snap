// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > render Card component 1`] = `
<div>
  <div
    class="cursor-pointer hover:scale-110 transition-all "
  >
    <img
      alt="Death of a Unicorn"
      src="https://image.tmdb.org/t/p/w500//lXR32JepFwD1UHkplWqtBP1K79z.jpg"
    />
    <div
      class="flex justify-between"
    >
      <p>
        Death of a Unicorn
      </p>
      <p>
        ‚≠ê
        6.4
      </p>
    </div>
  </div>
</div>
`;
